iroh-dns-server/src/store/signed_packets.rs

```diff
@@ -86,22 +133,28 @@ impl Actor {
                     Some(msg) = self.recv.recv() => {
                         match msg {
                             Message::Get { key, res } => {
+                                trace!("get {}", key);
                                 let packet = get_packet(&tables.signed_packets, &key)?;
                                 res.send(packet).ok();
                             }
                             Message::Upsert { packet, res } => {
                                 let key = PublicKeyBytes::from_signed_packet(&packet);
-                                let mut replaced = false;
-                                if let Some(existing) = get_packet(&tables.signed_packets, &key)? {
+                                trace!("upsert {}", key);
+                                let replaced = if let Some(existing) = get_packet(&tables.signed_packets, &key)? {
                                     if existing.more_recent_than(&packet) {
                                         res.send(false).ok();
                                         continue;
                                     } else {
-                                        replaced = true;
+                                        // remove the packet from the update time index
+                                        tables.update_time.remove(&packet.timestamp().to_be_bytes(), key.as_bytes())?;
```

Comment:
should this be `tables.update_time.remove(&existing.timestamp().to_be_bytes(), key.as_bytes())?;` (`existing` instead of `packet`)?